src/VisualStudio/Core/Def/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -133,9 +135,16 @@ private VisualStudioRemoteHostClientProvider(
             {
```

Comment:
(leaving comment here because I can't do it elsewhere)

I like the "wait until launched" mechanism you're adding for uses like the SolutionChecksumUpdater, since it means that as a generic unrelated piece of code can wait. What I don't get though is why we're not just passing the telemetry settings here when we're creating the process, rather than assuming that other thing is going to run?

Reply:
> What I don't get though is why we're not just passing the telemetry settings here when we're creating the process

i guess i don't understand.  how would we get those telemetry settings?  it seems like we need to be told about them.  which is when we go and pusha ll fo this to the OOP side.